lib/floki/selector_parser.ex

```diff
@@ -157,28 +141,34 @@ defmodule Floki.SelectorParser do
   defp do_parse_pseudo_not([{:close_parentesis, _} | t], pseudo_class) do
     {t, pseudo_class}
   end
+  defp do_parse_pseudo_not([{:space, _} | t], pseudo_class) do
+    do_parse_pseudo_not(t, pseudo_class)
+  end
   defp do_parse_pseudo_not(tokens, pseudo_class) do
-    {after_parentesis, before_parentesis} = split_list_for_not_tokens(tokens, [])
-
-    selector = before_parentesis
-               |> Enum.reverse
-               |> parse
-
-    if selector.combinator do
-      Logger.warn("Only simple selectors are allowed in :not() pseudo-class. Ignoring.")
-      {after_parentesis, nil}
-    else
-      {after_parentesis, %{pseudo_class | value: selector}}
-    end
+    do_parse_pseudo_not(tokens, %Selector{}, pseudo_class)
   end
-
-  defp split_list_for_not_tokens([{:close_parentesis, _} | t], tokens_before) do
-    {t, tokens_before}
+  defp do_parse_pseudo_not([{:close_parentesis, _} | t], pseudo_not_selector, pseudo_class) do
+    pseudo_class = update_pseudo_not_value(pseudo_class, pseudo_not_selector)
+    {t, pseudo_class}
+  end
+  defp do_parse_pseudo_not([{:comma, _} | t], pseudo_not_selector, pseudo_class) do
+    pseudo_class = update_pseudo_not_value(pseudo_class, pseudo_not_selector)
+    do_parse_pseudo_not(t, pseudo_class)
+  end
+  defp do_parse_pseudo_not([{:space, _} | t], pseudo_not_selector, pseudo_class) do
+    do_parse_pseudo_not(t, pseudo_not_selector, pseudo_class)
   end
-  defp split_list_for_not_tokens([], tokens_before) do
-    {[], tokens_before}
+  defp do_parse_pseudo_not([next_token | t], pseudo_not_selector, pseudo_class) do
+    pseudo_not_selector = do_parse([next_token], pseudo_not_selector)
```

Comment:
This seemed cleaner to parse the next token one by one within the parenthesis of `:not()` building all selectors within before leaving the pseudo_not parsing. It also avoids having to change any `do_parse` logic and trying to handle quite a few different edge cases